app/src/main/java/com/kickstarter/services/KSApolloClient.kt

```diff
@@ -548,88 +549,74 @@ class KSApolloClient(val service: ApolloClient) : ApolloClientType {
         }
     }
 }
-private fun backingObjectFromId(backingGr: GetBackingQuery.Backing?): Backing {
 
-    val name = backingGr?.backer()?.name() ?: ""
+private fun createBackingObject(backingGr: fragment.Backing?): Backing {
+    val payment = backingGr?.paymentSource()?.fragments()?.payment()?.let { payment ->
+        Backing.PaymentSource.builder()
+                .state(payment.state().toString())
+                .type(payment.type().rawValue())
+                .paymentType(CreditCardPaymentType.CREDIT_CARD.rawValue())
+                .id(payment.id())
+                .expirationDate(payment.expirationDate())
+                .lastFour(payment.lastFour())
+                .build()
+    }
+
     val id = decodeRelayId(backingGr?.id())?.let { it } ?: 0
-    val locationId = decodeRelayId(backingGr?.location()?.id())
-    val projectId = decodeRelayId(backingGr?.id()) ?: 0
-    val backerId = decodeRelayId(backingGr?.backer()?.id()) ?: 0
-    val rewardId = decodeRelayId(backingGr?.reward()?.id()) ?: 0
-    val avatar = Avatar.builder()
-            .medium(backingGr?.backer()?.imageUrl())
-            .build()
-    val rewardAmount = backingGr?.reward()?.amount()?.amount()?.toDouble()
-    val rewardSingleLocation = Reward.SingleLocation.builder()
-            .localizedName(backingGr?.location()?.displayableName())
-            .id(decodeRelayId(backingGr?.location()?.id())?: 0)
-            .build()
 
-    val reward = Reward.builder()
-            .minimum(rewardAmount?: 0.0)
-            .description(backingGr?.reward()?.description())
-            .estimatedDeliveryOn(DateTime(backingGr?.reward()?.estimatedDeliveryOn()))
-            .shippingSingleLocation(rewardSingleLocation)
-            .id(rewardId)
+    val location = backingGr?.location()?.fragments()?.location()
+    val locationId = decodeRelayId(location?.id())
+    val projectId = decodeRelayId(backingGr?.project()?.fragments()?.project()?.id()) ?: 0
+    val shippingAmount = backingGr?.shippingAmount()?.fragments()
+
+    val reward = backingGr?.reward()?.fragments()?.reward()?.let { reward ->
+        val rewardId = decodeRelayId(reward.id()) ?: 0
```

Comment:
I would coalesce this to `-1` just in case. There is some historical behaviour where a `Reward` with and ID of `0` is treated as "No Reward" 😄

Reply:
got it!